numpy/lib/function_base.py

```diff
@@ -3541,7 +3541,7 @@ def _percentile(a, q, axis=None, out=None,
     elif interpolation == 'higher':
         indices = ceil(indices).astype(intp)
     elif interpolation == 'midpoint':
-        indices = floor(indices) + 0.5
+        indices = 0.5 * (floor(indices) + ceil(indices))
```

Comment:
OK, this is what I get for reading this out of context -- indices is the location of the values that enter into the percentiles, not the calculated result. So this looks good to me.